### PR TITLE
Add nodes to DocumentFragments before attaching

### LINKED ISSUE #6246: v15-rc1 seems to skip some text nodes in IE
The following 

```
var Hello = React.createClass({
  render: function() {
    return <div><p>Hello</p>test</div>;
  }
});
```

results in only "Hello" being written in IE11 - [jsfiddle](https://jsfiddle.net/hL0jpazq/2/)

I realize there has been some changes to the way text nodes work in v15, but before we update our code to use `<span>`, I wanted to check if this is the intended behavior (since "test" is written in chrome).

### PR DESCRIPTION
**Pull request by zpao**
This is based on a quick chat with @spicyj. I haven't tested in IE11 yet (setting that up now), but after ensuring Firefox is running in lazy mode here, this does do what Ben was thinking, at least for the insert case. I need to do some more testing and see if this is an issue with updates (I think we should be ok there but will check).

Fixes #6246.

### CODE REVIEW

**Comment by spicyj** [comment]
Can we do .nodeType instead? That's how we check types everywhere else.

**Comment by spicyj** [comment]
Remove debugger too. ;)

**Comment by zpao** [comment]
Confirmed the test case does work now in IE11. Should we only do this if `enableLazy` is true @spicyj? Is it going to make much of a difference?

**Comment by spicyj** [comment]
Doesn't matter much since insertTreeChildren is a noop if enableLazy is false.

**Comment by zpao** [comment]
👍 I also tested Edge which seemed to be hitting this too (test case: http://playground.zpao.com/tmp/react_6246/)

File: src/renderers/dom/client/utils/DOMLazyTree.js [hunk 1]
```diff
@@ -53,8 +53,16 @@ function insertTreeChildren(tree) {
 
 var insertTreeBefore = createMicrosoftUnsafeLocalFunction(
   function(parentNode, tree, referenceNode) {
-    parentNode.insertBefore(tree.node, referenceNode);
-    insertTreeChildren(tree);
+    // Document Fragments in IE11, Edge (and possibly others) won't update
```
Comment (spicyj) [inline src/renderers/dom/client/utils/DOMLazyTree.js]: Sorry if it was unclear: this is how doc fragments work in all browsers. But we insert things in a different order in IE11 and Edge which is why they exhibit this bug.

Comment (gaearon) [inline src/renderers/dom/client/utils/DOMLazyTree.js]: Would running test with a mocked "is IE" flag solve this issue?

Comment (spicyj) [inline src/renderers/dom/client/utils/DOMLazyTree.js]: Yeah, we should run jest tests with enableLazy on. :\

Comment (zpao) [inline src/renderers/dom/client/utils/DOMLazyTree.js]: Comment updated: 5a17a1ef1d77f0a99cb708adde283275b7eb49fd
